File: In-memory-database/Level 2/Python/record.py
# record.py

class Record:
    def __init__(self):
        self.fields = {}  # field -> value

    def set_field(self, field: str, value: str):
        self.fields[field] = value

    def get_field(self, field: str) -> str:
        return self.fields.get(field, "")

    def delete_field(self, field: str) -> bool:
        if field in self.fields:
            del self.fields[field]
            return True
        return False

    def scan(self) -> str:
        return self._format_fields(self.fields)

    def scan_by_prefix(self, prefix: str) -> str:
        filtered = {
            field: val for field, val in self.fields.items() if field.startswith(prefix)
        }
        return self._format_fields(filtered)

    def _format_fields(self, fields: dict[str, str]) -> str:
        return ", ".join(f"{field}({fields[field]})" for field in sorted(fields))

File: In-memory-database/Level 2/C#/record.cs
using System.Collections.Generic;
using System.Linq;

public class Record
{
    private readonly Dictionary<string, string> _fields = new();

    public void SetField(string field, string value)
    {
        _fields[field] = value;
    }

    public string GetField(string field)
    {
        return _fields.TryGetValue(field, out var value) ? value : "";
    }

    public bool DeleteField(string field)
    {
        return _fields.Remove(field);
    }

    public string Scan()
    {
        return FormatFields(_fields);
    }

    public string ScanByPrefix(string prefix)
    {
        var filtered = _fields
            .Where(kv => kv.Key.StartsWith(prefix))
            .ToDictionary(kv => kv.Key, kv => kv.Value);
        return FormatFields(filtered);
    }

    private string FormatFields(Dictionary<string, string> fields)
    {
        return string.Join(", ", fields
            .OrderBy(kv => kv.Key)
            .Select(kv => $"{kv.Key}({kv.Value})"));
    }
}
